lib/python/qmk/cli/doctor.py

```diff
@@ -330,31 +337,41 @@ def doctor(cli):
     if bin_ok:
         cli.log.info('All dependencies are installed.')
     else:
-        ok = False
+        status = 'error'
 
     # Make sure the tools are at the correct version
+    ver_ok = []
     for check in (check_arm_gcc_version, check_avr_gcc_version, check_avrdude_version, check_dfu_util_version, check_dfu_programmer_version):
-        if not check():
-            ok = False
+        ver_ok.append(check())
+
+    if 'error' in ver_ok:
+        status = 'error'
+    elif 'warning' in ver_ok and status == 'ok':
+        status = 'warning'
 
     # Check out the QMK submodules
     sub_ok = check_submodules()
 
-    if sub_ok:
+    if sub_ok == 'ok':
         cli.log.info('Submodules are up to date.')
     else:
         if yesno('Would you like to clone the submodules?', default=True):
             submodules.update()
             sub_ok = check_submodules()
 
-        if not sub_ok:
-            ok = False
+        if 'error' in sub_ok:
+            status = 'error'
+        elif 'warning' in sub_ok and status == 'ok':
+            status = 'warning'
 
     # Report a summary of our findings to the user
-    if ok:
+    if status == 'ok':
```

Comment:
Instead of using strings like ok/warning/error for the status (which are typo prone, easy to mis type), I'd suggest using a python enum.

So add something like this at the beginning:
```py
from enum import Enum

class CheckStatus(Enum):
    OK = 1
    WARNING = 2
    ERROR = 3
```
Then at the end of your checks, return `CheckStatus.OK` for example, and chevk the same way `if status is CheckStatus.OK` for example

Reply:
Thanks for the suggestion.
I was thinking about doing this, but for some reason I decided against it. :shrug: 
As it turned out, your comment was the push I needed. :)